File: 0053-maximum-subarray/0053-maximum-subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        # using divide and conquer

        max_sum = -999999
        cur_sum = 0
        
        for i in range(len(nums)):
            cur_sum = cur_sum + nums[i]
            max_sum = max(max_sum, cur_sum)
            if cur_sum<0:
                cur_sum = 0
        
        return max_sum

File: 0053-maximum-subarray/0053-maximum-subarray.go
func maxSubArray(nums []int) int {
    curSum := 0
    maxSum := -math.MaxInt-1
    
    for _, elem := range nums{
        curSum += elem
        maxSum = max(maxSum, curSum)
        if curSum < 0{
            curSum = 0
        }
    }
    
    return maxSum
}